index.js

```diff
@@ -1 +1,207 @@
+    response.render('sighting', sightingData);
+  });
+});
+
+// Render a form to edit a sighting
+app.get('/sighting/:id/edit', (request, response) => {
+  // id of sighting is not zero-indexed
+  const { id } = request.params;
+  read(FILENAME, (data) => {
+    const sightingData = data.sightings[id - 1];
+    response.render('editExistingSightingForm', sightingData);
+  });
+});
+
+// Edit the sighting through a PUT method
+app.put('/sighting/:id/edit', (request, response) => {
+  // id of sighting is not zero-indexed
+  const { id } = request.params;
+  const requestBodyObj = convertToIterableObj(request.body);
+
+  // locally scoped boolean to track if current form is valid
+  let isThisFormValid = true;
+
+  // Perform input validation first
+  performValidationOnRequestBody(requestBodyObj, (feedbackMsg, areFieldsValid) => {
+    if (areFieldsValid === false) {
+      // Assigning feedback & id from helper function into request.body
+      request.body.feedback = feedbackMsg;
+      // Note that request.body generated by all the forms does not include id automatically,
+      // hence the inclusion
+      request.body.id = id;
+      isThisFormValid = false;
+      response.render('sightingFormEdit', request.body);
+    }
+  });
+  if (isThisFormValid === false) {
+    return;
+  }
+  /**
+ * @param {FILENAME} some file to be read
+ * @param {id} the corresponding id of the sighting
+ * @param {input} in this case, it is the iteratable requestBodyObj
+ * @param {callback} performs a callback after taking in the data and error vars from add fn
+*/
+  edit(FILENAME, id, requestBodyObj, (data, error) => {
+    if (error) {
+      response.sendStatus(500, 'error');
+      return;
+    }
+    response.redirect(`/sighting/${data.sightings.length}`);
+  });
+});
+
+// Render a list of all sighting shapes
+app.get('/shapes', (request, response) => {
+  read(FILENAME, (data) => {
+    response.render('allShapes', data);
+  });
+});
+
+// Render sighting based on a shape
+app.get('/shapes/:shape', (request, response) => {
+  const { shape: selectedShape } = request.params;
+  read(FILENAME, (data) => {
+    // Declare an object that stores the desired shape, followed by corresponding sighting
+    const selectedShapeSightings = { sightings: [], shape: '' };
+    selectedShapeSightings.shape = selectedShape;
+    // Filter by shape and stores all the associated sighting(s)
+    // into an sightings key/variable within selectedShapeSightings obj
+    selectedShapeSightings.sightings = data.sightings.filter((sighting) => sighting.shape === selectedShape);
+    response.render('shape', selectedShapeSightings);
+  });
+});
+
+// Accept a request to delete a sighting
+app.delete('/sighting/:id/', (request, response) => {
+  const { id } = request.params;
+  deleteContent(FILENAME, id, (data, error) => {
+    if (error) {
+      response.sendStatus(500, 'error');
+      return;
+    }
+    response.redirect('/');
+  });
+});
+
+app.listen(PORT);
```

Comment:
Thorough and great declaration of routes with relatively comprehensive commenting 👏